src/main/java/core/basesyntax/service/data/CsvDataServiceImpl.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.service.data;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.dao.FruitDaoImpl;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+import java.util.List;
+import java.util.Map;
+
+public class CsvDataServiceImpl implements DataService<String> {
```

Comment:
This service responsible for parsing data from String to FruitTransaction and for processing transactions with handlers, this violates SPR. Lets create separate service for processing data from String to FruitTransaction, and separate service where we will process transactions with operation handlers.
And all class dependencies should be injected through  CsvDataServiceImpl.class constructor (FruitDao, operationHandlerMap)